go/app/infra.go

```diff
@@ -20,30 +30,159 @@ type Item struct {
 //go:generate go run go.uber.org/mock/mockgen -source=$GOFILE -package=${GOPACKAGE} -destination=./mock_$GOFILE
 type ItemRepository interface {
 	Insert(ctx context.Context, item *Item) error
+	List(ctx context.Context) ([]Item, error) //商品一覧を取得するためのメソッドを追加
+	Get(ctx context.Context, id string) (*Item, error)
 }
 
 // itemRepository is an implementation of ItemRepository
 type itemRepository struct {
 	// fileName is the path to the JSON file storing items.
 	fileName string
+	// filePath is the absolute path to the JSON file
+	filePath string
 }
 
 // NewItemRepository creates a new itemRepository.
 func NewItemRepository() ItemRepository {
-	return &itemRepository{fileName: "items.json"}
+	cwd, err := os.Getwd()
+	if err != nil {
+		// エラーハンドリングを簡略化、パニックを使用
+		panic(fmt.Errorf("failed to get working directory: %w", err))
```

Comment:
We probably shouldn't panic as it will just crash the server instead of letting the user know that there is a server error.